testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/WelcomeScreenTest.java

```diff
@@ -141,7 +143,7 @@ public void resourcesTest() {
 
         // test my resources appeared
         accountWelcomeScreen.navigateTo();
-        accountWelcomeScreen.assertMyResourcesCardVisible(true);
+        Retry.execute(() -> accountWelcomeScreen.assertMyResourcesCardVisible(true), 10, 500);
```

Comment:
I think the correct fix would be either to use [`WaitUtils.waitUntilElement`](https://github.com/keycloak/keycloak/blob/2ddfc9449563cd703724b960a20d2bc000a68eb6/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/WaitUtils.java#L62-L64), or maybe even tweak a bit [`waitForPageToLoad`](https://github.com/keycloak/keycloak/blob/2ddfc9449563cd703724b960a20d2bc000a68eb6/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/WaitUtils.java#L163). It might also help other flaky tests.

Reply:
I got no clue how that stuff works, can you take a stab at that?

Reply:
You can reproduce the issues with just adding the following to ThemeResource#getResource:
```
try {
            Thread.sleep(500);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
```

Reply:
@vmuzikar I tried the waitForPageToLoad, but didn't work for me. I also tried extending the time it waits, but didn't actually see any difference in the amount of time it takes to run the test when it fails.

Reply:
Will try to look into it tomorrow. If my memory serves me well, `waitForPageToLoad` is the thing to tweak as it should assure the page is fully loaded after navigating to it.

Reply:
I've tried with WaitUtils.waitForPageToLoad, but it doesn't work. Honestly I don't think it's worth spending more time on this, unless you can find a simple way to do it.

Reply:
I think I found the issue. The waitings was not sometimes triggered properly. Trying an alternative fix [here](https://github.com/vmuzikar/keycloak/commit/da75ae71ab299adfe609e467841a34e8e7c8593e). Running 10 test runs with it [here](https://github.com/vmuzikar/keycloak/actions/runs/4074596544).

Reply:
Ok, my fix clearly [did not work](https://github.com/vmuzikar/keycloak/actions/runs/4074596544) and I can't spend more time on it right now. Feel free to merge this. :)